src/plugins/data/common/index_patterns/index_patterns/ensure_default_index_pattern.ts

```
@@ -47,24 +45,40 @@ export const createEnsureDefaultIndexPattern = (
     if (canUpdateUiSetting === false) {
       return;
     }
-    const patterns = await this.getIds();
-    let defaultId = await uiSettings.get('defaultIndex');
-    let defined = !!defaultId;
-    const exists = includes(patterns, defaultId);
-
-    if (defined && !exists) {
-      await uiSettings.remove('defaultIndex');
-      defaultId = defined = false;
-    }
-
+    const defaultId = await uiSettings.get('defaultIndex');
+    const defined = !!defaultId;
+    const patterns: string[] = [];
     if (defined) {
-      return;
+      const indexPattern = await this.get(defaultId);
+      const dataSourceRef = indexPattern?.dataSourceRef;
```

Nit: `if (!dataSourceRef) { return ; }`, prefer to fail fast.